server/registry/names/api.go

```diff
@@ -28,12 +28,16 @@ type Api struct {
 // Validate returns an error if the resource name is invalid.
 // For backward compatibility, names should only be validated at creation time.
 func (a Api) Validate() error {
+	if err := validateID(a.ApiID); err != nil {
+		return err
+	}
+
 	r := apiRegexp()
 	if name := a.String(); !r.MatchString(name) {
 		return fmt.Errorf("invalid API name %q: must match %q", name, r)
```

Comment:
Unfortunately, I think this error message is the same as [this one](https://github.com/apigee/registry/blob/616e3a92c71fbb528430f649700215f8ec3c9f06/server/registry/names/api.go#L101).